File: backend/src/main/java/com/financeapp/service/UserProfileService.java
package com.financeapp.service;

import com.financeapp.dto.user.UpdateProfileRequest;
import com.financeapp.dto.user.UserProfileResponse;
import java.util.UUID;

// TODO: view/update profile (fullName, email) - reuse AuthService.changePassword for password changes.
public interface UserProfileService {
    UserProfileResponse getProfile(UUID userId);
    UserProfileResponse updateProfile(UUID userId, UpdateProfileRequest request);
}

File: src/services/userProfileService.ts
import api from './api'

export interface UserProfile {
  id: string
  fullName: string
  email: string
  createdAt: string
}

export const userProfileService = {
  async get() {
    const { data } = await api.get<UserProfile>('/profile')
    return data
  },
  async update(fullName: string) {
    const { data } = await api.put<UserProfile>('/profile', { fullName })
    return data
  }
}
